htlcswitch/interfaces.go

```diff
@@ -33,8 +33,9 @@ type InvoiceDatabase interface {
 	// passed payment hash.
 	CancelInvoice(payHash lntypes.Hash) error
 
-	// SettleHodlInvoice settles a hold invoice.
-	SettleHodlInvoice(preimage lntypes.Preimage) error
+	// SettleHodlInvoice settles a hold invoice. If the preimage isn't known
+	// yet, it can be passed in to this call.
+	SettleHodlInvoice(hash lntypes.Hash, preimage *lntypes.Preimage) error
```

Comment:
Why do we also need to drag along the hash given it can be derived from the pre-image? Or is it that this is for the case where you know the invoice only by hash and we can derive the pre-image at runtime via some other mechanism?

Reply:
This is for the case where `lnd` already knows the preimage and it is sufficient to reference the invoice only by hash.